File: applicatie/src/main/java/com/vdzon/administratie/rest/factuur/FactuurResource.java
package com.vdzon.administratie.rest.factuur;

import com.vdzon.administratie.util.JsonUtil;
import spark.Spark;

import javax.inject.Inject;


public class FactuurResource {

    @Inject
    public FactuurResource(FactuurService factuurService) {
        Spark.put("/rest/factuur/", factuurService::putFactuur, JsonUtil.json());
        Spark.post("/rest/factuur/", factuurService::putFactuur, JsonUtil.json());
        Spark.delete("/rest/factuur/:uuid", factuurService::removeFactuur, JsonUtil.json());
        Spark.get("/rest/factuur/pdf/:uuid", factuurService::getPdf, JsonUtil.json());
    }


}

File: rest/src/main/java/com/vdzon/administratie/rest/factuur/FactuurResource.kt
package com.vdzon.administratie.rest.factuur

import com.vdzon.administratie.util.JsonUtil
import spark.Route
import spark.Spark

import javax.inject.Inject


class FactuurResource
@Inject
constructor(factuurService: FactuurService) {
    init {
        Spark.put("/rest/factuur/", Route { req, res -> factuurService.putFactuur(req, res) }, JsonUtil.json())
        Spark.post("/rest/factuur/", Route { req, res -> factuurService.putFactuur(req, res) }, JsonUtil.json())
        Spark.delete("/rest/factuur/:uuid", Route { req, res -> factuurService.removeFactuur(req, res) }, JsonUtil.json())
        Spark.get("/rest/factuur/pdf/:uuid", Route { req, res -> factuurService.getPdf(req, res) }, JsonUtil.json())
    }


}
